sdk/scheduler/src/main/java/com/mesosphere/sdk/state/PersistentOperationRecorder.java

```diff
@@ -34,7 +36,13 @@ public void record(Operation operation, Offer offer) throws Exception {
     private void recordTasks(List<Protos.TaskInfo> taskInfos) throws StateStoreException {
         logger.info(String.format("Recording %d updated TaskInfos/TaskStatuses:", taskInfos.size()));
         List<Protos.TaskStatus> taskStatuses = new ArrayList<>();
-        for (Protos.TaskInfo taskInfo : taskInfos) {
+        Collection<Protos.TaskInfo> unpackedTaskInfos;
+        try {
+            unpackedTaskInfos = CommonTaskUtils.unpackTaskInfos(taskInfos);
```

Comment:
Oh right, well, that's the idea anyway -- PersistentOfferRecorder would retrieve the original unpacked TaskInfos directly from LaunchOfferRecommendation